### DISCUSSION TITLE
The condition doesn't work, anyone can help check?

### DISCUSSION THREAD

**Discussion by Neal86** [AutomaApp/automa#1109] [Q&A]
HI, everyone, I create two conditions, and tried many times to let it works as I want, but both of them doesn't work, any one can help check?
Condition 1: 
path1: if the tags(https://prnt.sc/gOmLg3zpVouD) contain keywords"cat" or "smallcat", it will go to condition 2(https://prnt.sc/Y-kipDGYRlFe).
false:  it will go to click the next video button(https://prnt.sc/HY64uowcwNU4). 

condition 2: 
[TK condition.zip](https://github.com/AutomaApp/automa/files/10442504/TK.condition.zip)

path 1: If the follow button shows follow(https://prnt.sc/rUhpYxnOzA0i), it will go to click the follow button(https://prnt.sc/uhouardrKnxu), 
path 2: If the like button didn't be clicked yet(https://prnt.sc/T_0klEoZWjnk), it will go to click the like button(https://prnt.sc/dzOYd84cmTM7）
false: it will go to click the next video button(https://prnt.sc/k2qKP0CUBSgQ)

The workflow file
[TK condition.zip](https://github.com/AutomaApp/automa/files/10442506/TK.condition.zip)

**Comment by simtuzi1** [comment]
I don't have tictoc, but something like this should work.
Condition 1 could be:

``` javascript
const tagsElem = [...document.querySelectorAll(
  'div[class*="DivMainContent"] a[href*="/tag"]'
)];
const keywords = ['cat', 'smallcat'];
if (!tagsElem) {
  return false;
}
else if (tagsElem.some((el) => keywords.includes(el.innerText))){
console.log("now it works, this message is shown when condition is true");
}
```
I did not check for anything else. I included changes so you can try to import json and try it right away. 
I simplified the blocks a bit and changed first condition to be in while loop block. Although I'm not sure if that makes any difference.
I'm on a workplace computer and sometimes zip files are broken here, but aforementioned way should still work.

[cats-automa.zip](https://github.com/AutomaApp/automa/files/10479359/cats-automa.zip)

**Reply by Neal86** [reply]
THank you for your reply, i add your code and did a test, but the workflow keep runing the condition without end(https://prnt.sc/E9Dx2baDv1PM), can you help check?

**Reply by simtuzi1** [reply]
> THank you for your reply, i add your code and did a test, but the workflow keep runing the condition without end(https://prnt.sc/E9Dx2baDv1PM), can you help check?

Yes. I noticed that it's not working... I made some changes and actually got it to work. I'll update it after my work.

Basically I removed whole, condition block and replaced it with js code block
changed if statements to switch
below is the javascript block, which was before block with condition 1.
put right block id's to nextBlockId. Not sure if block id's are different on different computers (condition 2 is here "tm24li2" and fallback delayy is "vclyset"). Do not draw lines from javascript block to next block.

``` javascript
const tagsElem = [...document.querySelectorAll(
  'div[class*="DivMainContent"] a[href*="/tag"]'
)];
const keywords = ['cat', 'smallcat'];

  switch (tagsElem.some((el) => keywords.includes(el.innerText))){
    case true:
      automaNextBlock({ insert: false }, { nextBlockId: 'BLOCKIDHERETRUE' });
      break;
    case false:
      automaNextBlock({ insert: false }, { nextBlockId: 'BLOCKIDHEREFALSE' });
      break;
  }
```
